digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala

```diff
@@ -0,0 +1,609 @@
+
+    laneCode
+  }
+
+  /**
+    * @param newIncomeLane The lanes that will be updated
+    * @param linkIds  If only 1 means the user are change a specific lane and it can be cut or properties update
+    *                 If multiple means the chain is being updated and only properties are change ( Lane size / cut cannot be done)
+    * @param sideCode To know the direction
+    * @param username Username of the user is doing the changes
+    * @return
+    */
+  def update ( newIncomeLane: Seq[NewIncomeLane], linkIds: Set[Long] ,sideCode: Int, username: String ): Seq[Long] = {
+    if (newIncomeLane.isEmpty || linkIds.isEmpty)
+      return Seq()
+
+    withDynTransaction {
+     if (linkIds.size == 1) {
+        val linkId = linkIds.head
+
+        newIncomeLane.map { lane =>
+
+          val laneCode = getLaneCode(lane)
+
+          val lameToUpdate = PersistedLane(lane.id, linkId, sideCode, laneCode.toInt, lane.municipalityCode,
+            lane.startMeasure, lane.endMeasure, Some(username), None, None, None, false, 0, None, lane.attributes)
+
+          dao.updateEntryLane(lameToUpdate, username)
+        }
+
+      } else {
+
+        // get all lane codes from lanes to update
+        val laneCodes = newIncomeLane.map( lane => getLaneCode(lane).toInt )
+
+        // fetch from db the existing lanes
+        val allExistingLanes = dao.fetchLanesByLinkIdsAndLaneCode(linkIds.toSeq, laneCodes )
+                                  .groupBy(_.linkId)
+
+        // Iterate through the links
+        val result = linkIds.map { linkId =>
+          newIncomeLane.map { lane =>
+
+            val laneCode = getLaneCode(lane).toInt
+
+            val currentLanes = allExistingLanes.getOrElse(linkId, throw new InvalidParameterException(s"LinkId: $linkId dont have laneCode: $laneCode for update!"))
+            val auxLane = currentLanes.filter(_.laneCode == laneCode)
+
+            // Since cut lanes will not be in multiple linkIds Update, we can only have the 1st element
+            val currentLane = if ( auxLane.nonEmpty)
+                                auxLane.head
+                              else
+                                throw new InvalidParameterException(s"LinkId: $linkId dont have laneCode: $laneCode for update!")
+
+
+            val laneToUpdate = PersistedLane(currentLane.id, linkId, sideCode, laneCode, currentLane.municipalityCode,
+              currentLane.startMeasure, currentLane.endMeasure, Some(username), None, None, None, false, 0, None, lane.attributes)
+
+            dao.updateEntryLane(laneToUpdate, username)
+          }
+        }
+
+        result.head
+      }
+    }
+  }
+
+  def updatePersistedLanes ( lanes: Seq[PersistedLane], username: String ): Seq[Long] = {
+    if ( lanes.isEmpty )
+      return Seq()
+
+    withDynTransaction {
+      val result = lanes.map { lane =>
+        dao.updateEntryLane(lane, username)
+      }
+
+      if (result.isEmpty)
+        Seq()
+      else
+        result
+    }
+  }
```

Comment:
This should be enough 
```scala
 def updatePersistedLanes(lanes: Seq[PersistedLane], username: String): Seq[Long] = {
    withDynTransaction {
      lanes.map { lane =>
        dao.updateEntryLane(lane, username)
      }
    }
  }
```

Reply:
I think we can change for that:

```scala

def updatePersistedLanes ( lanes: Seq[PersistedLane], username: String ): Seq[Long] = {
    if ( lanes.isEmpty )
      return Seq()

    withDynTransaction {
      lanes.map { lane =>
        dao.updateEntryLane(lane, username)
      }
  }
```

Reply:
If lanes are empty then the .map will send a Seq() so I think the first if you put there is not needed

Reply:
apply as requested.